src/action/payment.js

```diff
@@ -199,12 +232,21 @@ class PaymentAction {
    */
   async estimateFee() {
     const { payment, settings } = this._store;
+    const amtSat = toSatoshis(payment.amount, settings);
+    const feeSat = await this._getFeeEstimateSat({
+      addr: payment.address,
+      amtSat,
+    });
+    payment.fee = toAmount(feeSat, settings);
+  }
+
+  async _getFeeEstimateSat({ addr, amtSat }) {
     const AddrToAmount = {};
-    AddrToAmount[payment.address] = toSatoshis(payment.amount, settings);
+    AddrToAmount[addr] = amtSat;
     const { feeSat } = await this._grpc.sendCommand('estimateFee', {
       AddrToAmount,
     });
-    payment.fee = toAmount(feeSat, settings);
+    return feeSat;
```

Comment:
Why do we need to create a new `_getFeeEstimateSat` helper function? Why not read `toSatoshis(payment.fee)`?

Reply:
lol fixed, yeah this was too complicated